salto-io/salto
# SALTO-4038 Added missing references handler in Jira

1. In this PR I added missing References resolver. 
2. I added enable missing references flag to the config.
3. I added Tests to test the missing references resolver. 

---

_Additional context for reviewer_
### To test the code manually
1. in the `jira.nacl` file wrote
``` 
    exclude = [
      {
      type = "Screen"
      }
    ]
``` 
2. fetch your jira env
3. Go to an instance of 'screen scheme' 
4. check that the screen reference is of missing reference type.  

### key notes
There is a flag called `enableMissingReferences`. When this flag set to false we do not convert missing references to references.

---
_Release Notes_: 
_Jira Adapter_:
When there is one object that points to another object and it is not found, instead of a reference that contains an internal id, there will be a missing reference which will emphasize that there is a missing reference problem. Each fetch will be accompanied by a warning and it will not be possible to deploy.

---
_User Notifications_: 
_Jira Adapter_:
1. When there are missing references, instead of meaningless numbers/names, a missing reference will appear whose purpose is to highlight the fact that there is a problem
